### DISCUSSION TITLE
API to list Repositories (and filter ?)

### DISCUSSION THREAD

**Discussion by bnpinel** [argoproj/argo-cd#5120] [Q&A] [answered]
Hi everyone,

I am having problem with REPOSITORIES api.  I am not sure if I am using it wrong or not.
In the doc, it is mentioned a query param : 
![image](https://user-images.githubusercontent.com/11931999/103069391-94a76300-45bf-11eb-957f-060f9156f795.png)

So I am trying to use the API as :
`GET /api/v1/repositories?repo=https%3A%2F%2Fgit_lab_url%2Fmy%2Frepo.git`

=> This is getting me the list of all available repos (but I am definitively positive that the repo exists).

Browsing issues led me to this #5004. And is mentioned a different way of using the API : 

`Perform API query by requesting this url: https://<argocd-server>/api/v1/repositories/<urlencoded form of repo url>`

With this, result is not the same : 
`GET /api/v1/repositories/https%3A%2F%2Fgit_lab_url%2Fmy%2Frepo.git`

=> This is getting me a 405 Methode Not Allowed.

What am I doing wrong ?

Thanks a lot for you answers.

**Comment by jannfis** [comment]
Tested against Argo CD running locally, with authentication disabled:

```
$ curl -s http://127.0.0.1:4000/api/v1/repositories/https%3A%2F%2Fgithub.com%2Fargoproj%2Fargocd-example-apps.git | jq -M
{
  "repo": "https://github.com/argoproj/argocd-example-apps",
  "connectionState": {
    "status": "Successful",
    "message": "",
    "attemptedAt": "2020-12-27T15:28:30Z"
  },
  "type": "git"
}
```

Are you sure there was no typo in your HTTP method?

**Reply by bnpinel** [reply]
Hello jannfis,

Thanks for your answer.

I am pretty sure the HTTP method is good, following is the curl request I am sending : 

curl --location --request GET 'https://ARGO_CD_URL/api/v1/repositories/https%3A%2F%2FMY_GIT_REPO%2Fsome%2Frepo%2Fsample-app%2Fkustomize.git' --header 'Authorization: Bearer SOME_TOKEN'

(Authentication is activated on my ArgoCD instance)

But now that I know that the request  should be working, I'll dig in with backend logs of ArgoCD, maybe I missed something during my investigation.

Once again, thanks a lot for your reply, I'll update the thread if I find anything.

**Comment by bnpinel** [comment, marked as answer]
Hello,

Something I forgot to mension, we are using Argo 1.5.3.
We managed to update to 1.8 and it's working as intended.

So seems to be a fixed bug :)

Thanks a lot for your answers and have a nice day !
